### PR TITLE
Create pagination component and add homepage

### PR DESCRIPTION
**Pull request by salih18**
-Create pagination component
-Create pagination states, actions, reducers, hooks
-Add Pagination component for homepage

### CODE REVIEW

File: client/src/components/Pagination.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+  };
+  // don't display pagination if there is only 1 page
+  if (!pages || pages.length <= 1) {
+    return null;
+  }
+  return (
+    <PaginationBootstrap className={`${paginationClass}`}>
+      <PaginationBootstrap.First onClick={() => setPage(1)} />
+      <PaginationBootstrap.Prev
+        onClick={() => {
```
Comment (sarah-vanderlaan) [inline client/src/components/Pagination.js:95]: I like to extract onClick handlers into functions if they are more than one line.  This helps with making the code really clear and easy to follow what is happening when the user clicks this component

An example:

```
const handleNavigatePreviousPage = () => {
    setPaginationReset(false);
    setPage(activePage - 1);
}
...
...
 <PaginationBootstrap.Prev
        onClick={handleNavigatePreviousPage}
        ....
```

File: client/src/components/Pagination.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+const Pagination = ({
+  items,
+  totalPages,
+  paginationClass,
+  setActivePage,
+  isSearchActive,
+  activePage,
+  paginationReset,
+  setPaginationReset,
```
Comment (sarah-vanderlaan) [inline client/src/components/Pagination.js:13]: when dealing with booleans, I suggest prefixing them with `is` or `should` to clearly show they are a boolean value

```suggestion
  shouldResetPagination,
  setShouldResetPagination,
```

File: client/src/components/Pagination.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        return { page: startPage + i, value: false };
+      }
+      if (i === 6) {
+        return { page: totalPages, value: true };
+      }
+    });
+
+    return pages;
+  };
+  // don't display pagination if there is only 1 page
```
Comment (sarah-vanderlaan) [inline client/src/components/Pagination.js:87]: Niceee! 😁

File: client/src/pages/proverb/ProverbList.js [hunk 4]
```diff
@@ -1,37 +1,71 @@
...
+import FlexTable from "../../components/FlexTable";
+import Pagination from "../../components/Pagination";
+import { useProverb, useSearch, usePagination } from "./../../redux/hooks";
+import { homepageTableTitle } from "./../../helpers/flexTableData";
 
 const ProverbList = () => {
-  const { proverbs, getProverbs } = useProverb()
-  const { filtered, isActive, searchTerm, setSearch } = useSearch()
+  const { proverbs, getProverbs, totalPages } = useProverb();
+  const { filtered, isActive, searchTerm, setSearch } = useSearch();
```
Comment (sarah-vanderlaan) [inline client/src/pages/proverb/ProverbList.js:10]: I would rename this to be very explicit that it is the filtered list of proverbs

```suggestion
  const { filtered: filteredProverbs, isActive, searchTerm, setSearch } = useSearch();
```

maybe also could change line 9 to:

```
  const { proverbs: allProverbs, getProverbs, totalPages } = useProverb();

```


then in your code below it makes it very clear:

`items={isActive ? filteredProverbs : allProverbs}`

File: client/src/redux/reducers/pagination.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  SET_PAGINATION_SIZE,
+  SET_PAGINATION_RESET,
+  CLEAR_PROFILE,
+  LOCATION_CHANGED,
+} from "./../actions/types";
+
+const initialState = {
+  activePage: 1,
+  pageSize: 10,
+  pageOfItems: [],
```
Comment (sarah-vanderlaan) [inline client/src/redux/reducers/pagination.js:12]:
```suggestion
  pageItems: [],
```

**Review by sarah-vanderlaan — COMMENTED** [review]
Awesome job @salih18!! 

I added some suggestions, let me know if you need any further explanation or clarification or if you have any questions/comments :)

File: client/src/components/Pagination.js [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+        } else {
+          setPage(activePage);
+        }
+      }
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [activePage, isSearchActive, items, setPage]
+  );
+
+  const createPages = () => {
```
[the remark this answers is not in the document]
Comment (salih18) [reply client/src/components/Pagination.js:44]: Thank you for your suggestion @sarah-vanderlaan . I like it. :)

File: client/src/components/Pagination.js (hunk 1 above)
Comment (salih18) [reply client/src/components/Pagination.js:95]: Ok. Let me use one handler function for all.

**Review by sarah-vanderlaan — APPROVED** [review]

**Pull request merged**